nixos/modules/services/web-apps/immich.nix

```
@@ -161,6 +164,17 @@ in
       };
     };
 
+    accelerationDevices = mkOption {
+      type = types.listOf types.str;
+      default = [ "*" ];
+      example = [ "/dev/dri/renderD128" ];
+      description = ''
+        A list of device paths to hardware acceleration devices that immich should
+        have access to. This is useful when transcoding media files.
+        The special value `"*"` will allow all devices.
```

```suggestion
        The special value `null` will disallow all devices using `PrivateDevices`.
```
would make more sense I think. In this case `[ ]` would mean access to all devices.

---

> In this case `[ ]` would mean access to all devices.

Is that how systemd handles it? That's kinda weird. And kinda confusing. Maybe put this explicitly in the docs too. That `[ ]` would allow access to all devices.

---

> Is that how systemd handles it?

No, this is a NixOS quirk. If you assign a list to an option in `serviceConfig`, it is converted to multiple lines. So the only way to disallow all devices is `DeviceAllow = [ "" ]` or `DeviceAllow = [ ]`. If you grep for `[ "" ]` you'll see many instances of this pattern.